WiseBuy/Managers/APIManager.m

```diff
@@ -47,10 +50,49 @@ + (void)fetchDealsFromEbayAPI:(NSString *)barcode {
         NSLog(@"%@", error);
         dispatch_semaphore_signal(sema);
       } else {
-//            NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *) response;
             NSError *parseError = nil;
             NSDictionary *responseDictionary = [NSJSONSerialization JSONObjectWithData:data options:0 error:&parseError];
-            NSLog(@"%@",responseDictionary);
+          
+          NSDictionary *finalResponseDic = responseDictionary[@"findItemsByProductResponse"][0][@"searchResult"][0];
+          NSArray *offerLists = finalResponseDic[@"item"];
+          for (NSDictionary* offer in offerLists)
+          {
+              NSString *price = offer[@"sellingStatus"][0][@"convertedCurrentPrice"][0][@"__value__"];
```

Comment:
Same here - always want to check if a value exists at that index before accessing